src/runtime/hexagon_remote/remote.cpp

```diff
@@ -21,6 +21,57 @@ extern "C" {
 typedef halide_hexagon_remote_handle_t handle_t;
 typedef halide_hexagon_remote_buffer buffer;
 
+#define MAX_WORKER_THREADS 4
+#define NUM_WORKER_THREADS_TO_CREATE 3
+#define STACK_SIZE 4096
+#define MAX_WORKLOADS 10
+static char stack [MAX_WORKER_THREADS][STACK_SIZE] __attribute__ ((__aligned__(128)));
+qurt_sem_t wait_for_work;
+
+struct work {
+    work *next_job;
+    int (*f)(void *, int, uint8_t *);
+    void *user_context;
+    int next, max;
+    uint8_t *closure;
+    int active_workers;
+    int exit_status;
+    qurt_cond_t wakeup_owner;
+    // A job can be in the following states.
+    // claimed - When the thread_pool has started work on the entire job, but not necessarily
+    //           completed it. Condition: next >= max.
+    // running - When there are active workers on the job. Condition: active_workers > 0;
+    // done - When the job is completely done and there are no active workers on the job.
+    bool claimed() { return next >= max; }
+    bool running() { return active_workers > 0; }
+    bool done() { return claimed() && !running(); }
+};
+bool thread_pool_initialized;
```

Comment:
Seems like this should get initialized to false? I think maybe there's something in C that says globals get zero initialized but I think it would be good to explicitly initialize it.